### ISSUE TITLE
Populate events api with fields and implement proper permissions

### ISSUE THREAD

**Issue by shubham-padia** [fossasia/open-event-orga-server#3693]
Events api has been setup for the purpose of tickets. Populate events api with relevant fields excluding relationships like sessions etc whose api has not been setup currently.

**Comment by LuD1161** [comment]
@shubham-padia @niranjan94 
I am trying to send this JSON 
```
{
	"data":{
		"type":"event",
		"attributes":{
			  "name": "name_of_event",
			  "email": "email@domain.com",
			  "state": "Draft",
			  "starts_at": "2017-06-03 18:45:00+05:30",
			  "ends_at": "2017-06-30 18:45:00+05:30",
			  "longitude": 0,
			  "latitude": 0,
			  "large": "http://website.com",
			  "description": "string",
			  "privacy": "public"
			}
		}
}
```
to http://localhost:5000/v1/events [ POST ]

**It mentions null value in `starts_at` column , but I am passing it .
Isn't it the correct way to do it ?**

and am getting the following error :

```
{
  "errors": [
    {
      "status": 500,
      "source": {
        "pointer": "/data"
      },
      "detail": "Object creation error: (psycopg2.IntegrityError) null value in column \"starts_at\" violates not-null constraint\nDETAIL:  Failing row contains (12, 6734113a, name_of_event, null, null, null, null, null, UTC, null, null, null, null, , null, null, null, null, null, 1, , f, public, Draft, null, null, null, null, , null, f, null, null, null, null, f, f, f, f, f, f, null, null, null, 2017-06-01 06:59:20.49354, null, null, null, f, null).\n [SQL: 'INSERT INTO events (identifier, name, event_url, email, logo, starts_at, ends_at, timezone, latitude, longitude, location_name, searchable_location_name, description, background_url, thumbnail, large, icon, organizer_name, show_map, organizer_description, has_session_speakers, privacy, state, type, topic, sub_topic, ticket_url, code_of_conduct, schedule_published_on, ticket_include, deleted_at, payment_country, payment_currency, paypal_email, tax_allow, pay_by_paypal, pay_by_stripe, pay_by_cheque, pay_by_bank, pay_onsite, cheque_details, bank_details, onsite_details, created_at, pentabarf_url, ical_url, xcal_url, sponsors_enabled, discount_code_id) VALUES (%(identifier)s, %(name)s, %(event_url)s, %(email)s, %(logo)s, %(starts_at)s, %(ends_at)s, %(timezone)s, %(latitude)s, %(longitude)s, %(location_name)s, %(searchable_location_name)s, %(description)s, %(background_url)s, %(thumbnail)s, %(large)s, %(icon)s, %(organizer_name)s, %(show_map)s, %(organizer_description)s, %(has_session_speakers)s, %(privacy)s, %(state)s, %(type)s, %(topic)s, %(sub_topic)s, %(ticket_url)s, %(code_of_conduct)s, %(schedule_published_on)s, %(ticket_include)s, %(deleted_at)s, %(payment_country)s, %(payment_currency)s, %(paypal_email)s, %(tax_allow)s, %(pay_by_paypal)s, %(pay_by_stripe)s, %(pay_by_cheque)s, %(pay_by_bank)s, %(pay_onsite)s, %(cheque_details)s, %(bank_details)s, %(onsite_details)s, %(created_at)s, %(pentabarf_url)s, %(ical_url)s, %(xcal_url)s, %(sponsors_enabled)s, %(discount_code_id)s) RETURNING events.id'] [parameters: {'pay_by_stripe': False, 'pay_by_bank': False, 'schedule_published_on': None, 'pay_by_cheque': False, 'topic': None, 'event_url': None, 'timezone': 'UTC', 'logo': None, 'deleted_at': None, 'searchable_location_name': None, 'discount_code_id': None, 'tax_allow': False, 'onsite_details': None, 'location_name': None, 'privacy': 'public', 'paypal_email': None, 'ical_url': None, 'background_url': None, 'state': 'Draft', 'latitude': None, 'organizer_description': u'', 'type': None, 'thumbnail': None, 'organizer_name': None, 'description': u'', 'payment_country': None, 'cheque_details': None, 'sponsors_enabled': False, 'ticket_url': None, 'ticket_include': False, 'payment_currency': None, 'pay_onsite': False, 'pentabarf_url': None, 'icon': None, 'ends_at': None, 'name': u'name_of_event', 'pay_by_paypal': False, 'created_at': datetime.datetime(2017, 6, 1, 6, 59, 20, 493540), 'longitude': None, 'email': None, 'large': None, 'has_session_speakers': False, 'sub_topic': None, 'starts_at': None, 'identifier': '6734113a', 'show_map': 1, 'bank_details': None, 'xcal_url': None, 'code_of_conduct': u''}]",
      "title": "Unknown error"
    }
  ],
  "jsonapi": {
    "version": "1.0"
  }
}
```
I have been following from here : [http://flask-rest-jsonapi.readthedocs.io/en/latest/quickstart.html#create-object](http://flask-rest-jsonapi.readthedocs.io/en/latest/quickstart.html#create-object)

**Comment by niranjan94** [comment]
@LuD1161 please take note of the change in requirement

**Comment by LuD1161** [comment]
Closed via [PR#3708](https://github.com/fossasia/open-event-orga-server/pull/3708)
